packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -293,6 +293,7 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
   };
 
   // Store the scroll position so we can restore it later.
+  /// TODO: should this happen all the time??
```

Comment:
I can't remember why we did this or what it fixed. Maybe it isn't needed anymore??

Reply:
So you can reproduce by getting rid of the `scrollPos` references completely and by using the "useTable -> Scroll testing" story. Keyboard nav into the table and focus any cell that isn't in the first row, then tab out and shift tab back into the table:
Screen Recording 2024-05-29 at 5.05.56 PM

If my memory + notes on the PR serves, this is due to how focus lands on the last checkbox in the table before being marshalled to the proper tracked focused key -> causes a scroll position change in the scrollable body -> messes up scrollIntoView calculations. Think the same issue happens if shift tabbing to a previously focused table column as well